docs/odh/llama-stack-with-odh.md

```diff
@@ -0,0 +1,171 @@
+- Cluster configured to use GPUs (g4dn.xlarge GPU nodes recommended)
+
+## Deploying vLLM via KServe
+
+### 1. Setup KServe in Standard Mode
+
+#### Configure DSCI
+In the DSCI resource, set the serviceMesh managementState to Removed:
+
+```yaml
+spec:
+  applicationsNamespace: redhat-ods-applications
+  monitoring: opendatahub
+    managementState: Managed
+    namespace: opendatahub
+  serviceMesh:
+    controlPlane:
+      metricsCollection: Istio
+      name: data-science-smcp
+      namespace: istio-system
+    managementState: Removed
+```
+
+#### Configure DSC
+In the DSC resource, configure the kserve component:
+
+```yaml
+spec:
+  components:
+    kserve:
+      defaultDeploymentMode: RawDeployment
+      RawDeploymentServiceConfig: Headed
+      managementState: Managed
+      serving:
+        managementState: Removed
+        name: knative-serving
+```
+
+Verify the setup by checking that `kserve-controller-manager` and `odh-model-controller` pods are running:
+
+```bash
+oc get pods -n opendatahub | grep -E 'kserve-controller-manager|odh-model-controller'
+```
+
+### 2. Deploy LLaMA 3.2 Model via KServe UI
+
+1. Create a Connection:
+   - Go to ODH dashboard -> Create Project -> Connections tab -> Create connection
+   - **Project Name:** llamastack
+   - **Connection Type:** URI - v1
+   - **Connection name:** llama-3.2-3b-instruct
+   - **URI:** `oci://quay.io/redhat-ai-services/modelcar-catalog:llama-3.2-3b-instruct` (5.9 GiB)
+
+2. Deploy the Model:
+   - Go to Models tab → Single-model serving platform → Deploy model
+   - **Model deployment name:** llama-3.2-3b-instruct
+   - **Serving runtime:** vLLM NVIDIA GPU ServingRuntime for KServe
+   - **Model server size:** Custom (1 CPU core, 10 GiB memory)
+   - **Accelerator:** NVIDIA GPU (1)
+   - **Additional serving runtime arguments:**
+     ```
+     --dtype=half
+     --max-model-len=20000
+     --gpu-memory-utilization=0.95
+     --enable-chunked-prefill
+     --enable-auto-tool-choice
+     --tool-call-parser=llama3_json
+     --chat-template=/app/data/template/tool_chat_template_llama3.2_json.jinja
+     ```
+
+3. Verify the model deployment:
+```bash
+oc get inferenceservice -n llamastack
+oc get pods -n llamastack | grep llama
+```
+
+### 3. Create LlamaStackDistribution CR
+
+```yaml
+apiVersion: llamastack.io/v1alpha1
+kind: LlamaStackDistribution
+metadata:
+  name: llamastack-custom-distribution
+  namespace: llamastack
+spec:
+  replicas: 1
+  server:
+    containerSpec:
+      env:
+        - name: INFERENCE_MODEL
+          value: llama-32-3b-instruct
+        - name: VLLM_URL
+          value: 'http://llama-32-3b-instruct-predictor.llamastack.svc.cluster.local:80/v1'
+        - name: MILVUS_DB_PATH
+          value: /.llama/distributions/remote-vllm/milvus.db
+      name: llama-stack
+      port: 8321
+    distribution:
+      image: 'quay.io/mcampbel/llama-stack:milvus-granite-embedding-125m-english'
```

Comment:
Just asking, is it okay to use an image from a user org?

Reply:
@Bobbins228 may have more context on this too

Reply:
@VaishnaviHire do you plan to push the image `quay.io/mcampbel/llama-stack:milvus-granite-embedding-125m-english` under the `redhat-et/llama` repo?

Reply:
I will defer to @cooktheryan for this.